packages/data-context/src/util/migration.ts

```diff
@@ -303,3 +271,69 @@ export function renameSpecPath (spec: string) {
   .replace('integration', 'e2e')
   .replace(/([._-]?[s|S]pec.|[.])(?=[j|t]s[x]?)/, '.cy.')
 }
+
+export function reduceConfig (cfg: OldCypressConfig): ConfigOptions {
+  const excludedFields = ['pluginsFile', '$schema', 'componentFolder']
+
+  return Object.entries(cfg).reduce((acc, [key, val]) => {
+    if (excludedFields.includes(key)) {
+      return acc
+    }
+
+    if (key === 'e2e' || key === 'component') {
+      const value = val as Record<string, unknown>
+
+      return { ...acc, [key]: { ...acc[key], ...value } }
+    }
+
+    if (key === 'testFiles') {
+      return {
+        ...acc,
+        e2e: { ...acc.e2e, specPattern: getSpecPattern(cfg, 'e2e') },
+        component: { ...acc.component, specPattern: getSpecPattern(cfg, 'component') },
+      }
+    }
+
+    if (key === 'ignoreTestFiles') {
+      return {
+        ...acc,
+        e2e: { ...acc.e2e, specExcludePattern: val },
+        component: { ...acc.component, specExcludePattern: val },
+      }
+    }
+
+    if (key === 'supportFile') {
+      return {
+        ...acc,
+        e2e: { ...acc.e2e, supportFile: val },
+        component: { ...acc.component, supportFile: val },
+      }
+    }
+
+    if (key === 'baseUrl') {
+      return {
+        ...acc,
+        e2e: { ...acc.e2e, [key]: val },
+      }
+    }
+
+    return { ...acc, global: { ...acc.global, [key]: val } }
+  }, { global: {}, e2e: {}, component: {} })
+}
+
+function getSpecPattern (cfg: OldCypressConfig, testType: TestingType) {
+  const specPattern = cfg[testType]?.testFiles ?? cfg.testFiles ?? '**/*.cy.js'
+  const customComponentFolder = cfg.component?.componentFolder ?? cfg.componentFolder ?? null
+
+  if (testType === 'component' && customComponentFolder) {
+    return `${customComponentFolder}/${specPattern}`
+  }
+
+  const customIntegrationFolder = cfg.e2e?.integrationFolder ?? cfg.integrationFolder ?? null
```

Comment:
https://github.com/cypress-io/cypress/blob/cec46dfc058a8745438060552729f158abc3f9a1/packages/data-context/src/sources/migration/shouldShowSteps.ts#L40

Reply:
I added these back because they kept giving me the default. Though I'm not 100% on what we should do if one isn't provided. Should it be `${testFiles}` or should it be `${defaultFolder}/${testFiles}`?